src/main/java/frc/robot/subsystems/ArmSubsystem.java

```diff
@@ -0,0 +1,219 @@
+     private DoublePublisher m_TargetAngle;
+     private DoublePublisher m_ArmPosPub;
+
+    //for  arm ff
+    private DoubleSubscriber momentToVoltageConversion;
+    private double m_VoltageConversion;
+
+    //spark absolute encoder
+    SparkAbsoluteEncoder m_AbsEncoder;  
+    //embedded relative encoder
+    //private RelativeEncoder m_bottomEncoder;
+    public SparkPIDController m_pidControllerArm;  
+
+    ProfiledPIDFFController m_profiledFFController = new ProfiledPIDFFController();
+
+    public static ArmSubsystem getInstance() {
+      if (instance == null) {
+        SubsystemChecker.subsystemConstructed(SubsystemType.ArmSubsystem);
+        instance = new ArmSubsystem();
+      }
+      return instance;
+    }
+    public ArmSubsystem() {
+      m_Arm = new CANSparkMax(Config.CANID.ARM_SPARK_CAN_ID, motorType); //creates SparkMax motor controller for bottom joint
+      ErrorCheck.errREV(m_Arm.restoreFactoryDefaults());
+      ErrorCheck.errREV(m_Arm.setSmartCurrentLimit(ArmConfig.CURRENT_LIMIT));
+      m_Arm.setInverted(ArmConfig.SET_INVERTED); //sets movement direction
+      ErrorCheck.errREV(m_Arm.setIdleMode(IdleMode.kBrake)); //sets brakes when there is no motion
+      
+      ErrorCheck.errREV(m_Arm.setSoftLimit(SoftLimitDirection.kForward, ArmConfig.arm_forward_limit));
+      ErrorCheck.errREV(m_Arm.setSoftLimit(SoftLimitDirection.kReverse, ArmConfig.arm_reverse_limit));
+      ErrorCheck.errREV(m_Arm.enableSoftLimit(SoftLimitDirection.kForward, ArmConfig.SOFT_LIMIT_ENABLE));
+      ErrorCheck.errREV(m_Arm.enableSoftLimit(SoftLimitDirection.kReverse, ArmConfig.SOFT_LIMIT_ENABLE));
+
+      m_Arm.setPeriodicFramePeriod(PeriodicFrame.kStatus5, 20);
+      m_Arm.setPeriodicFramePeriod(PeriodicFrame.kStatus6, 20);
+
+      m_AbsEncoder = m_Arm.getAbsoluteEncoder(Type.kDutyCycle);
+      m_AbsEncoder.setInverted(true);
+      m_AbsEncoder.setPositionConversionFactor(2*Math.PI);
+      m_AbsEncoder.setVelocityConversionFactor(2*Math.PI/60.0);
+      m_AbsEncoder.setZeroOffset(Math.toRadians(52));
+
+      m_pidControllerArm = m_Arm.getPIDController();
+      m_pidControllerArm.setFeedbackDevice(m_AbsEncoder);
+
+      
+    NetworkTable ArmTuningTable = NetworkTableInstance.getDefault().getTable(m_tuningTable);
+    m_ArmPSubs = ArmTuningTable.getDoubleTopic("P").getEntry(ArmConfig.arm_kP, PubSubOption.periodic(0.02));
+    m_ArmISubs = ArmTuningTable.getDoubleTopic("I").getEntry(ArmConfig.arm_kI);
+    m_ArmDSubs = ArmTuningTable.getDoubleTopic("D").getEntry(ArmConfig.arm_kD);
+    m_ArmIzSubs = ArmTuningTable.getDoubleTopic("IZone").getEntry(ArmConfig.arm_kIz);
+    m_ArmFFSubs = ArmTuningTable.getDoubleTopic("FF").getEntry(ArmConfig.arm_kFF);
+   //m_topArmOffset = topArmTuningTable.getDoubleTopic("Offset").getEntry(ArmConfig.top_arm_offset);
+    momentToVoltageConversion = ArmTuningTable.getDoubleTopic("VoltageConversion").subscribe(m_VoltageConversion);
+
+    m_ArmFFSubs.accept(ArmConfig.arm_kFF);
```

Comment:
Instead of accept, use setDefault here (the method name might be slightly different). That will ensure NT has a value but won't overwrite anything tuned in NT after redeploys.